src/imagewriter.cpp

```diff
@@ -334,19 +358,55 @@ void ImageWriter::startWrite()
 
     if (!_expectedHash.isEmpty() && _cachedFileHash == _expectedHash)
     {
-        // Check cache file integrity before using it
-        if (_verifyCacheFileIntegrity())
+        // Use background cache manager to check cache file integrity
```

Comment:
This might be a stupid question, but does it ignore all the cache stuff if you select a local already-downloaded `.img` file?